heat/core/manipulations.py

```diff
@@ -911,6 +911,268 @@ def hstack(tup):
+            )
+        # invalid list/tuple
+        elif not all(isinstance(r, int) for r in repeats):
+            raise TypeError(
+                "Invalid type within repeats. All components of repeats must be integers."
+            )
+        # valid list/tuple
+        else:
+            repeats = factories.array(
+                repeats, dtype=types.int64, is_split=None, device=a.device, comm=a.comm
+            )
+
+        # check `repeats` is not empty
+        if repeats.gnumel == 0:
+            raise ValueError("Invalid input for repeats. Repeats must contain data.")
+
+        # check `repeats` is 1-dimensional
+        if len(repeats.shape) != 1:
+            raise ValueError(
+                "Invalid input for repeats. Repeats must be a 1d-object or integer, but "
+                "was {}-dimensional.".format(len(repeats.shape))
+            )
+
+    # start of algorithm
+
+    if 0 in a.gshape:
+        return a
+
+    # Broadcast (via int or 1-element DNDarray)
+    if isinstance(repeats, int) or repeats.gnumel == 1:
+        if axis is None and a.split is not None and a.split != 0:
+            print(
+                "\n!!! WARNING !!!\nIf axis is None, a has to be split along axis 0 if distributed.\nSplit axis of a "
+                "will be changed from {} to 0.".format(a.split)
+            )
+            a.resplit_(0)
+        if isinstance(repeats, int):
+            repeated_array_torch = torch.repeat_interleave(a._DNDarray__array, repeats, axis)
+        else:
+            if repeats.split is not None:
+                print(
+                    "\n!!! WARNING !!!\nFor broadcast via array_like repeats, repeats must not be "
+                    "distributed.\nSplit axis of repeats "
+                    "will be changed from {} to None.".format(repeats.split)
+                )
+                repeats.resplit_(None)
+            repeated_array_torch = torch.repeat_interleave(
+                a._DNDarray__array, repeats._DNDarray__array, axis
+            )
+    # No broadcast
+    else:
+        # check if the data chunks of `repeats` and/or `a` have to be (re)distributed before call of torch function.
+
+        # UNDISTRIBUTED CASE (a not distributed)
+        if a.split is None:
+            if repeats.split is not None:
+                print(
+                    "\n!!! WARNING !!!\nIf a is undistributed, repeats also has to be undistributed.\nSplit axis of "
+                    "repeats will be changed from {} to None.".format(repeats.split)
+                )
+                repeats.resplit_(None)
+
+            # Check correct input
+            if axis is None:
+                # check matching shapes (repetition defined for every element)
+                if a.gnumel != repeats.gnumel:
+                    raise ValueError(
+                        "Invalid input. Sizes of flattened a ({}) and repeats ({}) are not same. "
```

Comment:
can you capitalize `a`? or put it in quotes? something to indicate that its an input parameter and not just the letter a

Reply:
For sure! I got confused about this as well :smile: